sktree/tree/_multiview.py

```diff
@@ -379,28 +392,48 @@ def _build_tree(
         else:
             SPLITTERS = DENSE_SPLITTERS
 
-        if self.apply_max_features_per_feature_set:
+        if isinstance(self._max_features_arr, (Integral, Real, str, type(None))):
+            max_features_arr_ = [self._max_features_arr] * self.n_feature_sets_
+            stratify_mtry_per_view = self.apply_max_features_per_feature_set
+        else:
+            if not isinstance(self._max_features_arr, (list, np.ndarray)):
+                raise ValueError(
+                    f"max_features must be an array-like, int, float, str, or None; "
+                    f"got {type(self._max_features_arr)}"
+                )
+            if len(self._max_features_arr) != self.n_feature_sets_:
+                raise ValueError(
+                    f"max_features must be an array-like of length {self.n_feature_sets_}; "
+                    f"got {len(self.max_features)}"
+                )
+            max_features_arr_ = self._max_features_arr
+            stratify_mtry_per_view = True
+
+        self.n_features_in_set_ = []
+        if stratify_mtry_per_view:
             # XXX: experimental
             # we can replace max_features_ here based on whether or not uniform logic over
             # feature sets
             max_features_per_set = []
             n_features_in_prev = 0
             for idx in range(self.n_feature_sets_):
+                max_features = max_features_arr_[idx]
+
                 n_features_in_ = self.feature_set_ends_[idx] - n_features_in_prev
                 n_features_in_prev += n_features_in_
-
-                if isinstance(self.max_features, str):
-                    if self.max_features == "sqrt":
-                        max_features = max(1, int(np.sqrt(n_features_in_)))
-                    elif self.max_features == "log2":
-                        max_features = max(1, int(np.log2(n_features_in_)))
-                elif self.max_features is None:
+                self.n_features_in_set_.append(n_features_in_)
+                if isinstance(max_features, str):
+                    if max_features == "sqrt":
+                        max_features = max(1, math.ceil(np.sqrt(n_features_in_)))
```

Comment:
If we use `math.ceil` we shouldn't need `max` anymore, as the feature number should always be positive and has a minimum of 1.